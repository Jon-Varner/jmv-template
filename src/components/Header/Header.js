import React from 'react';

const Header = React.memo(({ title }) => (
  <header>
    <h1>{title}</h1>
  </header>
));

export default Header;
