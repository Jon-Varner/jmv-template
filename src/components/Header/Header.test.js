import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import Header from './Header';

afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(<Header title="Hello World" />);
  expect(asFragment()).toMatchSnapshot();
});

it('inserts title into h1', () => {
  const { getByText } = render(<Header title="Hello World" />);
  expect(getByText('Hello World')).toBeInTheDocument();
});
