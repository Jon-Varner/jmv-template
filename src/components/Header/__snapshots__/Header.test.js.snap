// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <h1>
      Hello World
    </h1>
  </header>
</DocumentFragment>
`;
